command/alloc_status.go

```diff
@@ -325,120 +325,130 @@ func (c *AllocStatusCommand) outputTaskStatus(state *api.TaskState) {
 
 	size := len(state.Events)
 	for i, event := range state.Events {
-		formatedTime := formatUnixNanoTime(event.Time)
-
-		// Build up the description based on the event type.
-		var desc string
-		switch event.Type {
-		case api.TaskSetup:
-			desc = event.Message
-		case api.TaskStarted:
-			desc = "Task started by client"
-		case api.TaskReceived:
-			desc = "Task received by client"
-		case api.TaskFailedValidation:
-			if event.ValidationError != "" {
-				desc = event.ValidationError
-			} else {
-				desc = "Validation of task failed"
-			}
-		case api.TaskSetupFailure:
-			if event.SetupError != "" {
-				desc = event.SetupError
-			} else {
-				desc = "Task setup failed"
-			}
-		case api.TaskDriverFailure:
-			if event.DriverError != "" {
-				desc = event.DriverError
-			} else {
-				desc = "Failed to start task"
-			}
-		case api.TaskDownloadingArtifacts:
-			desc = "Client is downloading artifacts"
-		case api.TaskArtifactDownloadFailed:
-			if event.DownloadError != "" {
-				desc = event.DownloadError
-			} else {
-				desc = "Failed to download artifacts"
-			}
-		case api.TaskKilling:
-			if event.KillReason != "" {
-				desc = fmt.Sprintf("Killing task: %v", event.KillReason)
-			} else if event.KillTimeout != 0 {
-				desc = fmt.Sprintf("Sent interrupt. Waiting %v before force killing", event.KillTimeout)
-			} else {
-				desc = "Sent interrupt"
-			}
-		case api.TaskKilled:
-			if event.KillError != "" {
-				desc = event.KillError
-			} else {
-				desc = "Task successfully killed"
-			}
-		case api.TaskTerminated:
-			var parts []string
-			parts = append(parts, fmt.Sprintf("Exit Code: %d", event.ExitCode))
+		if event.DisplayMessage != "" {
```

Comment:
The only difference between the if and the else is the formattedDisplayMessage. Should we instead do:

```
msg := event.DisplayMessage
if msg == "" {
  msg = buildDisplayMessage(event)
}
formattedTime := formatUnixNanoTime(event.Time)
formattedDisplayMsg := fmt.Sprintf("%s|%s|%s", formattedTime, event.Type, msg)
events[size-i] = formattedDisplayMsg
```